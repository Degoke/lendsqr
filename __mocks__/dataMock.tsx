import UsersIcon from '../public/images/users_card_icon.svg'

export const mockCard = {
  image: UsersIcon,
  text: 'Users',
  amount: 100,
}

export const mockImage = UsersIcon

export const MockComponent = () => (
  <div data-testid="mockComponent">Hello World</div>
)

export const mockList = {
  title: 'settings',
  items: [
    {
      text: 'preferences',
      url: '',
      icon: UsersIcon,
    },
    {
      text: 'fees and pricing',
      url: '',
      icon: UsersIcon,
    },
    {
      text: 'audit logs',
      url: '',
      icon: UsersIcon,
    },
  ],
}
