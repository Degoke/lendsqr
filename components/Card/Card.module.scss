@use '../../styles/variables' as *;

.card {
  padding: 1rem 1rem 0.5rem;

  &__text {
    font-size: 14px;
    text-transform: uppercase;
  }

  &__number {
    font-weight: 600;
    font-size: 24px;
    color: $text-secondary;
  }
}
