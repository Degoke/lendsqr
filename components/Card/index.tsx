import Image, { StaticImageData } from 'next/image'
import Paper from '../Paper'
import styles from './Card.module.scss'

type Props = {
  image: StaticImageData
  text: string
  amount: number
}

export default function Card({ image, text, amount }: Props) {
  return (
    <Paper>
      <div className={styles.card}>
        <Image src={image} alt="" />
        <p className={styles.card__text}>{text}</p>
        <p className={styles.card__number}>{amount}</p>
      </div>
    </Paper>
  )
}
