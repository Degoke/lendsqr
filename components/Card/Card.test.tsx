import { render } from '@testing-library/react'
import Card from './index'
import { mockCard } from '../../__mocks__/dataMock'

it('renders Card component unchanged', () => {
  const { container } = render(
    <Card
      image={mockCard.image}
      text={mockCard.text}
      amount={mockCard.amount}
    />
  )
  expect(container).toMatchSnapshot()
})
