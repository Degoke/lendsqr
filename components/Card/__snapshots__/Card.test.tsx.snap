// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Card component unchanged 1`] = `
<div>
  <div
    class="paper undefined"
  >
    <div
      class="card"
    >
      <img
        alt=""
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
      <p
        class="card__text"
      >
        Users
      </p>
      <p
        class="card__number"
      >
        100
      </p>
    </div>
  </div>
</div>
`;
